util/compression.h

```diff
@@ -1482,6 +1482,35 @@ inline std::string ZSTD_TrainDictionary(const std::string& samples,
 #endif  // ZSTD_VERSION_NUMBER >= 10103
 }
 
+inline std::string ZSTD_FinalizeDictionary(
+    const std::string& samples, const std::vector<size_t>& sample_lens,
+    size_t max_dict_bytes) {
+  // ZDICT_finalizeDictionary is only stable since version 1.4.5
+#if ZSTD_VERSION_NUMBER >= 10405  // v1.4.5+
+  assert(samples.empty() == sample_lens.empty());
+  if (samples.empty()) {
+    return "";
+  }
+  std::string dict_data(max_dict_bytes, '\0');
+  size_t dict_len = ZDICT_finalizeDictionary(
+      dict_data.data(), max_dict_bytes, samples.data(),
+      static_cast<size_t>(samples.size()), samples.data(), sample_lens.data(),
```

Comment:
According to my reading of the API, it may want the first parameter (`dstDictBuffer`) to actually contain the raw samples content. Want to try that and see if it gives better results?

Reply:
Never mind, it looks like the samples content should be copied. Hm...

Reply:
However, there is something wrong with the way I'm calling this API: a few ZDICT errors make me realize that the third parameter (`dictContent`) should not be longer than `max_dict_bytes`. So using `samples` here as the third parameter is incorrect...